tests/test_async_client.py

```
@@ -0,0 +1,1736 @@
+        assert client.github_token is None
+        assert client.max_concurrent == 5
+        assert client.connector_limit == 10
+        assert client._session is None
+        assert client._semaphore is not None  # Semaphore is now created in __init__
+        assert client._closed is False
+        assert client._rate_limit_remaining == {}
+        assert client._rate_limit_reset == {}
+
+    def test_init_with_token(self):
+        """Test initialization with GitHub token."""
+        client = AsyncGitHubClient(github_token="ghp_test_token")
```

Ruff (0.15.0)

[error] Possible hardcoded password assigned to argument: "github_token"

(S106)